## ${\textbf{Import Libraries}}$

In [12]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import sklearn
import itertools
from pandas import Series, DataFrame
from pylab import rcParams
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn.model_selection import GridSearchCV , cross_val_score
from sklearn.metrics import classification_report , accuracy_score , roc_auc_score
from sklearn import (metrics,preprocessing)
import warnings
warnings.filterwarnings("ignore")
rcParams['figure.figsize'] = 10, 8
sns.set_style('whitegrid')

# Tasks

## 1.Introduction
This assignment will help you to consolidate the concepts learnt in the
session.


## 2.Problem Statement
Predicting Survival in the Titanic Data Set
We will be using a decision tree to make predictions about the Titanic data
set from Kaggle. This data set provides information on the Titanic
passengers and can be used to predict whether a passenger survived or
not.
Loading Data and modules


Url=
https://raw.githubusercontent.com/BigDataGal/Python-for-DataScience/master/titanic-train.csv
titanic = pd.read_csv(url)
titanic.columns =
['PassengerId','Survived','Pclass','Name','Sex','Age','SibSp','Parch','Ti
cket','Fare','Cabin','E mbarked']
You use only Pclass, Sex, Age, SibSp (Siblings aboard), Parch
(Parents/children aboard), and Fare to predict whether a passenger
survived.
NOTE:ThesolutionsharedthroughGithubshouldcontainthesource
codeusedand thescreenshotoftheoutput.
 
 
## 3. Output
This assignment consists of 200 marks and has to be submitted in
.ipynb/PDF format in the upcoming session for evaluation.

## ${\textbf{Import Data}}$

In [17]:
url = 'https://raw.githubusercontent.com/BigDataGal/Python-for-Data-Science/master/titanic-train.csv'

In [27]:
titanic = pd.read_csv(url)

In [23]:
len(titanic.columns)

12

## ${\textbf{Select Specific columns as tasked}}$

In [31]:
titanic.columns = ['PassengerId','Survived','Pclass','Name','Sex','Age','SibSp','Parch','Ticket','Fare','Cabin','E mbarked'] 

In [32]:
titanic.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin E mbarked  
0      0         A/5 21171   7.2500   NaN         S  
1      0          PC 17599  71.2833   C85         C  
2      0  STON/O2. 3101282   7.9250   NaN         S  
3      0            113803  53.1000  C123         S  
4      0            373450   8.0500   NaN         S

In [28]:
print('Titanic Data - rows:' , titanic.shape[0],'columns:', titanic.shape[1])

Titanic Data - rows: 891 columns: 12


# Checking for missing values

In [33]:
missingInfo = list(titanic.columns[titanic.isnull().any()])
missingInfo

['Age', 'Cabin', 'E mbarked']

# Showing number of missing values in each column

In [34]:
for col in missingInfo:
    missingNumber = titanic[titanic[col].isnull() == True].shape[0]
    print('Missing Number for Col {}: {}'.format(col , missingNumber))
    

Missing Number for Col Age: 177
Missing Number for Col Cabin: 687
Missing Number for Col E mbarked: 2


# Percentage of missing values in dataframe


In [45]:
for col in missingInfo:
    percentMissing = titanic[titanic[col].isnull() == True].shape[0] / titanic.shape[0]
    print('Col Percent Missing {}: {}'.format(col, percentMissing))
    

Col Percent Missing Age: 0.19865319865319866
Col Percent Missing Cabin: 0.7710437710437711
Col Percent Missing E mbarked: 0.002244668911335578


## Columns and missing values

In [30]:
total = titanic.isnull().sum().sort_values(ascending = False)
percent = (titanic.isnull().sum()/titanic.isnull().count()*100).sort_values(ascending = False)
pd.concat([total, percent], axis=1, keys=['Total', 'Percent']).transpose()


Cabin        Age  Embarked  Fare  Ticket  Parch  SibSp  Sex  \
Total    687.000000  177.00000  2.000000   0.0     0.0    0.0    0.0  0.0   
Percent   77.104377   19.86532  0.224467   0.0     0.0    0.0    0.0  0.0   

         Name  Pclass  Survived  PassengerId  
Total     0.0     0.0       0.0          0.0  
Percent   0.0     0.0       0.0          0.0

### Observation:

Cabin column has 77% of its data missing.

Age has 19% of its data missing.

Embarked column has 22% of its data missing.